src/network/networkConfig.ts

```diff
@@ -0,0 +1,162 @@
+ */
+export type HttpRequestBody = string;
+
+/** Represents an HTTP request. */
+export interface HttpRequest {
+  /** The {@link HttpRequestBody} to send to the server. */
+  body?: HttpRequestBody;
+  /**
+   * The HTTP Headers of the request.
+   * Entries are expected to have the HTTP header as the key and its string content as the value.
+   */
+  headers: Record<string, string>;
+  /** The HTTP method of the request. */
+  method: string;
+  /** The URL of the request. */
+  url: string;
+}
+
+/**
+ * Base64-encoded string representing the HTTP response body.
+ */
+export type HttpResponseBody = string;
+
+/** Represents an HTTP response. */
+export interface HttpResponse {
+  /** The {@link HttpRequestBody} of the response. */
+  body?: HttpResponseBody;
+  /**
+   * The HTTP Headers of the response.
+   * Entries are expected to have the HTTP header as the key and its string content as the value.
+   */
+  headers: Record<string, string>;
+  /** The corresponding request object of the response. */
+  request: HttpRequest;
+  /** The HTTP status code of the response. */
+  status: number;
+  /** The URL of the response. May differ from {@link HttpRequest.url} when redirects have happened. */
+  url: string;
+}
+
+/**
+ * The network API gives the ability to influence network requests.
+ * It enables preprocessing requests and processing responses.
+ */
+export interface NetworkConfig extends NativeInstanceConfig {
+  /**
+   * Called before an HTTP request is made.
+   * Can be used to change request parameters.
+   *
+   * @param type Type of the request to be made.
+   * @param request The HTTP request to process.
+   * @returns A Promise that resolves to an `HttpRequest` object.
+   *          - If the promise resolves, the player will use the processed request.
+   *          - If the promise rejects, the player will fall back to using the original request.
+   *
+   * @examples
+   * ```
+   *  const requestCallback = (type: HttpRequestType, request: HttpRequest) => {
+   *    // Access current properties
+   *
+   *    console.log(JSON.stringify(type));
+   *    console.log(JSON.stringify(request));
+   *
+   *    // Modify the request
+   *
+   *    request.headers['New-Header'] = 'val';
+   *    request.method = 'GET';
+   *
+   *    // Return the processed request via a Promise
+   *
+   *    const processed: HttpRequest = {
+   *      body: request.body,
+   *      headers: request.headers,
+   *      method: request.method,
+   *      url: request.url,
+   *    };
+   *    return Promise.resolve(processed);
+   *  };
+   *
+   *  const player = usePlayer({
+   *    networkConfig: {
+   *      preprocessHttpRequest: requestCallback,
+   *    },
+   *  });
+   * ```
+   */
+  preprocessHttpRequest?: (
+    type: HttpRequestType,
+    request: HttpRequest
+  ) => Promise<HttpRequest>;
+  /**
+   * Called before an HTTP response is accessed by the player.
+   * Can be used to the access or change properties of the response.
```

Comment:
1. There is an extra "the"
2. It can be simplified to "change the response"
```suggestion
   * Can be used to access or change the response.
```